### PR TITLE
image list output for builds

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by chengliangli0918 — APPROVED** [review]

**Comment by alexeldeib** [comment]
repoDigests seems empty for containerd even though locally testing the same images it works. not sure it matters since we still get image ID (although there are interesting quirks - https://github.com/distribution/distribution/issues/1662#issuecomment-213079540)

also this defaults to installing crictl latest version, should probably tweak that a bit more or use ctr instead

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 1]
```diff
@@ -119,7 +119,6 @@ installCrictl() {
         tar zxvf "$CRICTL_DOWNLOAD_DIR/${CRICTL_TGZ_TEMP}" -C ${CRICTL_BIN_DIR}
         chmod 755 $CRICTL_BIN_DIR/crictl
     fi
-    rm -rf ${CRICTL_DOWNLOAD_DIR}
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_install.sh:122]: why i do this?

**Comment by ganeshkumar5699** [comment]
> repoDigests seems empty for containerd even though locally testing the same images it works. not sure it matters since we still get image ID (although there are interesting quirks - [distribution/distribution#1662 (comment)](https://github.com/distribution/distribution/issues/1662#issuecomment-213079540))
> 
> also this defaults to installing crictl latest version, should probably tweak that a bit more or use ctr instead

Should ctr also be used for generating this list (and ctr's list seems to be a superset of the crictl list?) ? @sozercan had shared earlier about finding images through `ctr -n k8s.io images list` and also identified some SHAs in the image list (on the node) whose component was not clear.

File: vhdbuilder/packer/install-dependencies.sh [hunk 2]
```diff
@@ -153,6 +153,9 @@ if [[ ${CONTAINER_RUNTIME:-""} == "containerd" ]]; then
     downloadCrictl ${CRICTL_VERSION}
     echo "  - crictl version ${CRICTL_VERSION}" >> ${VHD_LOGS_FILEPATH}
   done
+  
+  KUBERNETES_VERSION=$(echo -e "$CRICTL_VERSIONS" | head -n 1) installCrictl || exit $ERR_CRICTL_DOWNLOAD_TIMEOUT
```
Comment (ganeshkumar5699) [inline vhdbuilder/packer/install-dependencies.sh:157]: Is the output stored in KUBERNETES_VERSION, being used elsewhere?

Comment (ganeshkumar5699) [inline vhdbuilder/packer/install-dependencies.sh:157]: Do we need to handle separate installation of `crictl` for docker vs containerd builds here?

**Comment by alexeldeib** [comment]
ah, need to recreate this pr from not-my-fork for security on PR checks

File: vhdbuilder/packer/install-dependencies.sh (hunk 2 above)
Comment (alexeldeib) [reply vhdbuilder/packer/install-dependencies.sh:157]: in installCrictl :)

Comment (alexeldeib) [reply vhdbuilder/packer/install-dependencies.sh:157]: shouldn't really matter since docker will use...docker I suppose

**Comment by alexeldeib** [comment]
moved to https://github.com/Azure/AgentBaker/pull/1985

**Pull request closed without merging**